docker/nvm-install.sh

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env bash
+
+{ # this ensures the entire script is downloaded #
+
+nvm_has() {
+  type "$1" > /dev/null 2>&1
+}
+
+if [ -z "$NVM_DIR" ]; then
+  NVM_DIR="$HOME/.nvm"
+fi
+
+nvm_latest_version() {
+  echo "v0.31.2"
+}
+
+#
+# Outputs the location to NVM depending on:
+# * The availability of $NVM_SOURCE
+# * The method used ("script" or "git" in the script, defaults to "git")
+# NVM_SOURCE always takes precedence unless the method is "script-nvm-exec"
+#
+nvm_source() {
+  local NVM_METHOD
+  NVM_METHOD="$1"
+  local NVM_SOURCE_URL
+  NVM_SOURCE_URL="$NVM_SOURCE"
+  if [ "_$NVM_METHOD" = "_script-nvm-exec" ]; then
+    NVM_SOURCE_URL="https://raw.githubusercontent.com/creationix/nvm/$(nvm_latest_version)/nvm-exec"
+  elif [ -z "$NVM_SOURCE_URL" ]; then
+    if [ "_$NVM_METHOD" = "_script" ]; then
+      NVM_SOURCE_URL="https://raw.githubusercontent.com/creationix/nvm/$(nvm_latest_version)/nvm.sh"
+    elif [ "_$NVM_METHOD" = "_git" ] || [ -z "$NVM_METHOD" ]; then
+      NVM_SOURCE_URL="https://github.com/creationix/nvm.git"
+    else
+      echo >&2 "Unexpected value \"$NVM_METHOD\" for \$NVM_METHOD"
+      return 1
+    fi
+  fi
+  echo "$NVM_SOURCE_URL"
+}
+
+nvm_download() {
+  if nvm_has "curl"; then
+    curl -q $*
```

Comment:
I think this script uses curl/wget as well. I don't think this will solve the issue unless we download the binary above and edit this download method

Reply:
Hmm I'm looking into this now, this pulls a script which pulls another script. The final script `nvm.sh` does also have `curl | sh` but I believe that's an update method and I think we could manually pull it out. It may take me a bit to sort this all out